app/controllers/projects_controller.rb

```diff
@@ -34,6 +34,11 @@ def show
     )
     @dates = events_array.uniq { |e| e["date"] }.map { |d| d["date"] }
     @events = Event.format_for_chart(events_array)
+
+    @total_count = 0
+    @events.each do |name, value|
+      @total_count += value[:total_count]
+    end
   end
```

Comment:
This is not very important, but as a learning exercise, could you try to do this with an `inject` (also called `reduce`)? See more about that method [here](https://apidock.com/ruby/Enumerable/inject).

Reply:
Furthermore, I suggest to move this method to a better place. Similiar to `Event.format_for_chart`, you could move it to the `Event` model since it's related to events.

Alternatively, you could leverage the existing `Event.format_for_chart` method to not only return @events, but also a separate variable, the @`total_count`. This might make sense since we can avoid an additional loop and the two variables are kind of related.

In ruby, you can return two variables as an array, and then assign them like so:
```ruby
@events, @total_count = Event.format_for_chart(events_array)
```

This is called destructuring, you can read more about it [here.](https://www.fullstackruby.dev/ruby-3-fundamentals/2021/01/06/everything-you-need-to-know-about-destructuring-in-ruby-3/)

Reply:
instead of:
``` ruby
@total_count = 0
@events.each do |name, value|
  @total_count += value[:total_count]
end
```
I can use `.inject`like this:
```ruby
@total_count = @events.values.inject(0) { |sum, e| sum + e[:total_count] }
```

Is that correct or do you have a better solution?

Reply:
Yep!